### DISCUSSION TITLE
Do you support LDAP proxying with overlays?

### DISCUSSION THREAD

**Discussion by gdonval** [kanidm/kanidm#3276] [Q&A] [answered]
The problem I'm in is that I have to "consume" data from a couple AD/LDAP servers that I don't control and are a mess. I want to be able to perform authorisation/existence check through normal binding and, hopefully, augment their records with Kanidm-local information.

Is this a use-case currently covered?

**Comment by yaleman** [comment]
I'd suggest looking at the [sync docs](https://kanidm.github.io/kanidm/stable/sync/concepts.html), it might possibly meet your needs.

**Reply by gdonval** [reply]
Thanks for the suggestion. I had seen that and I fully realise this is the best way forward but I was really wondering about a lighter-weight solution where only password authentication is delegated to an external LDAP. It's fine if there is none: that'd simply be something Kanidm cannot do at the moment.

**Comment by Firstyear** [comment, marked as answer]
Yes, sync is the best option here - you sync accounts and groups in Kanidm, and then kanidm can host their MFA and other details. The sync can be run for an extended period too. 

We currently don't support AD as a sync provider, but we'd be interested to have the support.

**Reply by gdonval** [reply]
A full sync is problematic and will be painful (for social reasons) when all I really need to be compliant is to use our existing LDAP server for auth (not necessarily for group membership).

E.g. Zitadel has a mode of operation where it uses LDAP as an authentication backend only, with metadata (like groups) being managed from within it. 

Would that be something you are interested in providing (or that you already provide)?

**Reply by yaleman** [reply]
Our intention/design is to be the identity _**provider**_, whereas Zitadel is an authentication broker, which are very different intentions/designs.

**Reply by Firstyear** [reply]
Yeah, we wouldn't be interested in this - to achieve our goals, we need to be the IDP, not a middleware.

**Reply by gdonval** [reply]
No worries, thanks a lot for the clarification!
